Sources/tart/Commands/VNC.swift

```diff
@@ -0,0 +1,89 @@
+import ArgumentParser
+import Foundation
+import Network
+import SystemConfiguration
+import NIO
+
+struct VNCCredentialsNotFound: Error {
+}
+
+struct VNC: AsyncParsableCommand {
+    static var configuration = CommandConfiguration(abstract: "Connect to the VM via VNC")
+
+    @Argument(help: "VM name")
+    var name: String
+
+    @Option(help: "Number of seconds to wait for the VNC credentials")
+    var wait: UInt16 = 0
+
+    @Flag(help: "Open instead of displaying an URL")
+    var open: Bool = false
+
+    func run() async throws {
+        do {
+            let vmDir = try VMStorageLocal().open(name)
+
+            var vncCredentials = try await Self.retrieveVNCCredentials(
+                    vncCredentialsURL: vmDir.vncCredentialsURL,
+                    secondsToWait: wait)
+
+            if vncCredentials == nil {
+                let vmConfig = try VMConfig(fromURL: vmDir.configURL)
+
+                if let ip = try await IP.resolveIP(vmConfig, secondsToWait: wait) {
+                    vncCredentials = VNCCredentials(host: ip.debugDescription)
+                }
+            }
+
+            if let vncCredentials = vncCredentials {
+                if open {
+                    vncCredentials.open()
+                } else {
+                    print(vncCredentials.url())
+                }
+
+                Foundation.exit(0)
+            }
+
+            print("no VNC credentials found, is your VM running?")
```

Comment:
I think this phrase is confusing if I just did `tart run macos` and then `open vnc:/<ip>`. What kind of credential are we talking about here?

Having a separate `vnc` command seems just make things more complex. We initially talked only about using the private APIs instead of the current implementation of `--vnc` flag for `run` command. Just to be able to VNC into a  VM crated from scratch.

Reply:
> Having a separate `vnc` command seems just make things more complex. We initially talked only about using the private APIs instead of the current implementation of `--vnc` flag for `run` command. Just to be able to VNC into a VM crated from scratch.

Isn't the whole point of using the private API is the ability to do automatic installs via Packer? Which in turn requires a `tart vnc` command to be able to query the VNC credentials programatically and connect to the VNC using them?